protocols/client/client.go

```diff
@@ -400,6 +401,7 @@ func HybridVSockDialer(sock string, timeout time.Duration) (net.Conn, error) {
 	}
 
 	dialFunc := func() (net.Conn, error) {
+		handshakeTimeout := 3 * time.Second
```

Comment:
I think this is not enough as it might introduce some potential errors if for some use cases (nested environment) it was really taking some time to get the handshake properly done.
I think 10s would give more slack.